src/tools/annotation/bidirectionalTool/moveHandle/movePerpendicularLine.js

```diff
@@ -0,0 +1,87 @@
+import external from './../../../../externalModules.js';
+import getDistance from './getDistance.js';
+import getLineVector from './getLineVector.js';
+import getBaseData from './getBaseData.js';
+
+function isFixedEnd(baseData) {
+  const { fixedPoint, perpendicularEnd } = baseData;
+
+  return fixedPoint === perpendicularEnd;
+}
+
+function getMultiplier(baseData) {
+  return isFixedEnd(baseData) ? -1 : 1;
+}
+
+function getMovingPoint(baseData) {
```

Comment:
For all of these methods, really, we should try to author more explicit signatures

Reply:
Changed all the methods and added separate files and unit test files for them.
Also documented them all with JSDoc.
Thank you!